tls/datadog_checks/tls/tls.py

```diff
@@ -0,0 +1,348 @@
+                tags=self._tags,
+                message='Unable to parse the certificate: {}'.format(e),
+            )
+            return
+
+        self.check_protocol_version(protocol_version)
+        self.validate_certificate(cert)
+        self.check_age(cert)
+
+    def check_local(self, instance):
+        if not self._local_cert_path:
+            raise ConfigurationError('You must specify `local_cert_path` in your configuration file.')
+        elif self._validate_hostname and not self._server_hostname:
+            raise ConfigurationError(
+                'You must specify `server_hostname` in your configuration file, or disable `validate_hostname`.'
+            )
+
+        try:
+            with open(self._local_cert_path, 'rb') as f:
+                cert = f.read()
+        except Exception as e:
+            self.service_check(
+                self.SERVICE_CHECK_VALIDATION,
+                self.CRITICAL,
+                tags=self._tags,
+                message='Unable to open the certificate: {}'.format(e),
+            )
+            return
+
+        try:
+            cert = self.local_cert_loader(cert)
+        except Exception as e:
+            self.service_check(
+                self.SERVICE_CHECK_VALIDATION,
+                self.CRITICAL,
+                tags=self._tags,
+                message='Unable to parse the certificate: {}'.format(e),
+            )
+            return
+
+        self.validate_certificate(cert)
+        self.check_age(cert)
+
+    def check_protocol_version(self, version):
+        if version in self._allowed_versions:
+            self.service_check(self.SERVICE_CHECK_VERSION, self.OK, tags=self._tags)
+        else:
+            self.service_check(
+                self.SERVICE_CHECK_VERSION,
+                self.CRITICAL,
+                tags=self._tags,
+                message='Disallowed protocol version: {}'.format(version),
+            )
+
+    def validate_certificate(self, cert):
+        if self._validate_hostname:
+            validator, host_type = self.validation_data
+
+            try:
+                validator(cert, text_type(self._server_hostname))
+            except service_identity.VerificationError:
+                self.service_check(
+                    self.SERVICE_CHECK_VALIDATION,
+                    self.CRITICAL,
+                    tags=self._tags,
+                    message='The {} on the certificate does not match the given host'.format(host_type),
+                )
+                return
+            except service_identity.CertificateError as e:  # no cov
+                self.service_check(
+                    self.SERVICE_CHECK_VALIDATION,
+                    self.CRITICAL,
+                    tags=self._tags,
+                    message='The certificate contains invalid/unexpected data: {}'.format(e),
+                )
+                return
+
+        self.service_check(self.SERVICE_CHECK_VALIDATION, self.OK, tags=self._tags)
```

Comment:
Should we `ok` if `self._validate_hostname` is `false`?

Reply:
Yes b/c it also send ok/critical for parsing errors